### ISSUE TITLE
ISSUE-52: Implement opendkim.diffheaders()

### ISSUE THREAD

**Issue by godsflaw** [godsflaw/node-opendkim#52]
Implement `opendkim.diffheaders()` to discern how signed header fields were changed in transit in order to debug verification problems.  This issue depends on #50.

- [ ] code
- [ ] tests
- [ ] documentation

For a good example of the above items see: https://github.com/godsflaw/node-opendkim/pull/33
Note that the first commit title should start with `ISSUE-52:`

more information:
http://www.opendkim.org/libopendkim/dkim_diffheaders.html

Bounty:
https://ethlance.com/#/job/242

**Comment by nabetse00** [comment]
Hello, I just found this bounty in Ethlance. I just wanted to make sure the bounty is still valid since it was posted a long time ago. If it is still valid I would be happy to complete it.

**Comment by godsflaw** [comment]
@nabetse00 yes, this job is still open.  Let me know if you have questions.

**Comment by nabetse00** [comment]
Yes a have some questions.

- How do you want to handle libopendkim recompilation (needed to get diffheader feature). On my dev box i recompiled the whole opendkim package with adecuate options.
- What kind of test do you require. I asume basic print of diffheaders, bad parameters and errors if initialization not done do you need more ?

**Comment by godsflaw** [comment]
hmm, now I remember this did require a recompile.  I would say, we need to make sure that it fails elegantly if this feature is not compiled in.  The good news is, we control the build environment where tests live.  In theory, rather than package installing `opendkim-dev` here:
https://github.com/godsflaw/node-opendkim/blob/dev/Dockerfile#L19-L21
We could either make a new Docker container as a base image that does all of that and compiles `opendkim` with the right options (this is probably the best route), or we could grab the source and compile it in this image (this would work, but will make test runs take forever).

For now, if you get the code working in your local environment and push a PR, I can then add onto that PR the `Dockerfile` changes and push the new base image.  Just toss me the links to where you grabbed the source and the flags you used for compilation in the PR and I'll work it up.

Sound good?

**Comment by nabetse00** [comment]
Sounds perfect !

The source code was grabbed from command line via apt-get source libopendkim-dev, then i modified the debian rule file to include the compile flags. I'll put the rule file and generated debs in a separate folder named package in the PR.

I think the PR will be ready tomorrow.

**Comment by nabetse00** [comment]
Hello @godsflaw i've added some memory leak test for `diffheaders` to my fork and adpated the `dockerfile` for `leakage` module instalation (memory leak package for node.js).

As expected no memory leak. If you are interested in including those changes i can make a new PR.

**Comment by godsflaw** [comment]
@nabetse00 sounds great, please do.
